tests/contrib/pymongo/test_pymongo_patch.py

```diff
@@ -22,10 +33,54 @@ class TestPymongoPatch(PatchTestCase.Base):
     __get_version__ = get_version
 
     def assert_module_patched(self, pymongo):
-        pass
+        self.assert_wrapped(pymongo.MongoClient.__init__)
+        self.assert_wrapped(Topology.select_server)
+
+        if _VERSION >= (3, 12):
+            self.assert_wrapped(Server.run_operation)
+        elif _VERSION >= (3, 9):
+            self.assert_wrapped(Server.run_operation_with_response)
+        else:
+            self.assert_wrapped(Server.send_message_with_response)
+
+        if _VERSION >= (4, 5):
+            self.assert_wrapped(Server.checkout)
+        else:
+            self.assert_wrapped(Server.get_socket)
+        self.assert_wrapped(Connection.command)
+        self.assert_wrapped(Connection.write_command)
 
     def assert_not_module_patched(self, pymongo):
-        pass
+        self.assert_not_wrapped(pymongo.MongoClient.__init__)
+        self.assert_not_wrapped(Topology.select_server)
+        if _VERSION >= (3, 12):
+            self.assert_not_wrapped(Server.run_operation)
+        elif _VERSION >= (3, 9):
+            self.assert_not_wrapped(Server.run_operation_with_response)
+        else:
+            self.assert_not_wrapped(Server.send_message_with_response)
+
+        if _VERSION >= (4, 5):
+            self.assert_not_wrapped(Server.checkout)
+        else:
+            self.assert_not_wrapped(Server.get_socket)
+
+        self.assert_not_wrapped(Connection.command)
+        self.assert_not_wrapped(Connection.write_command)
 
     def assert_not_module_double_patched(self, pymongo):
-        pass
+        self.assert_not_double_wrapped(Topology.select_server)
```

Comment:
Maybe a silly question, but do we not need to check whether `pymongo.MongoClient.__init__` is double wrapped too?

Reply:
Good catch thank you!